modules/server/src/main/ts/bedrock/util/ExecUtils.ts

```diff
@@ -0,0 +1,140 @@
+import * as which from 'which';
+import * as Arr from './Arr';
+import * as crossSpawn from 'cross-spawn';
+import { ChildProcess } from 'child_process';
+import * as http from 'http';
+import * as deepmerge from 'deepmerge';
+
+export interface ChildAPI {
+  start: (...args: any[]) => Promise<ChildProcess | null> | ChildProcess | null;
+  stop: () => void;
+  defaultInstance: ChildProcess | null;
+  readonly isNpm?: boolean;
+}
+
+export const execLoader = (exec: string, args: string[] = []): ChildAPI => {
+  const api = childAPIWrapper(crossSpawn, false, exec, args);
+  return api;
+};
+
+// Npm drivers (e.g. geckodriver) are called differently to binaries. This is to conform them
+// into a single compatible type. I don't think npm drivers were used until WDIOv8, which has them
+// installed by default as dependencies.
```

Comment:
That's correct, and it's actually quite bad for us. Having drivers as direct dependencies seems like a nice way to reduce the load on the ops team, until you realise that means browser upgrades require _every project to upgrade bedrock_, rather than a single server upgrade - the server upgrades have been painful, but upgrading 50 repositories would be more painful.

I suppose the premium monorepo gets to eat this lunch, as once it is ready there will only be two place to upgrade bedrock, but that's still painful for every active branch when a browser upgrade happens.

Even so, I figure one day we'll end up removing the non-npm driver code.